lib/dataContract/stateTransition/validation/validateDataContractSTDataFactory.js

```diff
@@ -40,6 +43,24 @@ function validateDataContractSTDataFactory(dataProvider, validateIdentityExisten
       );
     }
 
+    if (!result.isValid()) {
+      return result;
+    }
+
+    // Verify ST signature
+    const userId = stateTransition.getUserId();
+    const identity = await dataProvider.fetchIdentity(userId);
+    const { publicKey } = identity.getPublicKeyById(stateTransition.getPublicKeyId());
+    // eslint-disable-next-line no-underscore-dangle
+    const publicKeyHash = new PublicKey(Buffer.from(publicKey, 'base64'))._getID();
+    const signatureIsValid = stateTransition.verifySignature(publicKeyHash);
```

Comment:
Let's pass Identity Public Key model here

Reply:
We can have two methods for this, one that accepts just a buffer and returns a boolean for cases when the key cannot be obtained from the identity model, like in the case with registration, when we need to use a key from a different source than identity. And the second one that accepts identity's PublicKey model